bcl/src/bc_i2c.c

```diff
@@ -1,156 +1,142 @@
 #include <bc_i2c.h>
 #include <bc_module_core.h>
+#include <bc_tick.h>
 #include <stm32l0xx.h>
 
+#define _BC_I2C_MEMORY_ADDRESS_SIZE_8BIT    1
+#define _BC_I2C_MEMORY_ADDRESS_SIZE_16BIT   2
+#define _BC_I2C_RELOAD_MODE                I2C_CR2_RELOAD
+#define _BC_I2C_AUTOEND_MODE               I2C_CR2_AUTOEND
+#define _BC_I2C_SOFTEND_MODE               (0x00000000U)
+#define _BC_I2C_NO_STARTSTOP               (0x00000000U)
+#define _BC_I2C_GENERATE_START_WRITE       I2C_CR2_START
+
+static inline bool _bc_i2c_mem_write(I2C_TypeDef *I2Cx, uint8_t device_address, uint16_t memory_address, uint16_t memory_address_length, uint8_t *buffer, uint16_t length);
```

Comment:
Please remove all inline statements. Speaking the I2C communication there will be no CPU penalty. Otherwise this will result in excessive flash area usage...